pkg/dao/snapshots.go

```diff
@@ -18,16 +19,37 @@ func (sDao snapshotDaoImpl) Create(s *Snapshot) error {
 }
 
 // List the snapshots for a given repository config
-func (sDao snapshotDaoImpl) List(repoConfigUuid string) ([]Snapshot, error) {
+func (sDao snapshotDaoImpl) List(repoConfigUuid string, paginationData api.PaginationData, _ api.FilterData) (api.SnapshotCollectionResponse, int64, error) {
 	var snaps []Snapshot
+	var totalSnaps int64
 
-	result := sDao.db.Model(&Snapshot{}).
+	filteredDB := sDao.db
+	result := filteredDB.
 		Joins("inner join repository_configurations on repository_configurations.repository_uuid = snapshots.repository_uuid").
 		Where("repository_configurations.uuid = ?", repoConfigUuid).
 		Where("snapshots.org_id = repository_configurations.org_id").
-		Find(&snaps)
+		Limit(paginationData.Limit).
+		Offset(paginationData.Offset).
+		Find(&snaps).Count(&totalSnaps)
+	resp := snapshotConvertToResponses(snaps)
 	if result.Error != nil {
-		return snaps, result.Error
+		return api.SnapshotCollectionResponse{}, 0, result.Error
 	}
-	return snaps, nil
+	return api.SnapshotCollectionResponse{Data: resp}, totalSnaps, nil
+}
+
+// Converts the database models to our response objects
+func snapshotConvertToResponses(snapshots []Snapshot) []api.SnapshotResponse {
+	repos := make([]api.SnapshotResponse, len(snapshots))
+	for i := 0; i < len(snapshots); i++ {
+		snapshotModelToApi(snapshots[i], &repos[i])
+	}
+	return repos
+}
+
+func snapshotModelToApi(model Snapshot, resp *api.SnapshotResponse) {
+	resp.CreatedAt = model.CreatedAt
+	resp.DistributionPath = model.DistributionPath
+	resp.PackageCount = int(model.ContentCounts["rpm.package"])
+	resp.ErrataCount = int(model.ContentCounts["rpm.advisory"])
```

Comment:
i had kinda imagined we'd return ContentCounts as is (essentially a map string -> int).   This makes adding types in the future simpler.  Any Thoughts?

Reply:
Meaning the response looks like this?
```
{
   "created_at": [timestamp],
   "distribution_path": "aaa/bbb",
   "content_counts": {
        "rpm.package": 30,
        "rpm.advisory": 4,
        "rpm.packagegroup": 2,
         ...,
   }
}
```

Reply:
correct.  Its passing the key straight from pulp, which is kinda ugly, but its tucked into a nice little structure.  

If we wanted a cleaner key, we could translate from  'rpm.package' to 'package' or something similar.   I try to think about the future if we have some other type of repository that isn't yum

Reply:
Got it. I'm okay with returning it that way.